test/fw/ptl/utils/plugins/ptl_report_json.py

```diff
@@ -0,0 +1,155 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+import os
+import sys
+from ptl.utils.pbs_dshutils import DshUtils
+
+
+class PTLJsonData(object):
```

Comment:
PTLJsonData is not a Plugin itself, so I kind of feel that it should not be its own module inside the plugins directory. It contains only one class and it's only used internally inside the ptl_test_db module, so why not just move PTLJsonData to that module itself?

Reply:
There is a future plan to support test information generation in json format which needs this part of the code to be used by another plugin - ptl_test_info. Data base plugin ptl_test_db comes into picture only when test is executed; hence it is separated out of ptl_test_db plugin. Since it is a helper code for two plugins it is placed inside plugins directory.